commands/registry.go

```diff
@@ -974,3 +1038,201 @@ func RunGetRegistryOptionsRegions(c *CmdConfig) error {
+  - The registry name
+  - The endpoint used to access it
+  - The region where it resides
+`
+	cmdRunRegistriesGet := CmdBuilder(
+		cmd,
+		RunRegistriesGet, "get <registry-name>",
+		"Get information about a registry",
+		getRegistryDesc,
+		Writer, aliasOpt("g"), displayerType(&displayers.Registry{}),
+	)
+	cmdRunRegistriesGet.Example = `The following example retrieves information about a registry named ` + "`" + `example-registry` + "`" + `: doctl registries get example-registry`
+
+	deleteRegistryDesc := "Permanently deletes a registry from your account."
+	cmdRunRegistriesDelete := CmdBuilder(
+		cmd,
+		RunRegistriesDelete, "delete <registry-name>",
+		"Delete a registry",
+		deleteRegistryDesc,
+		Writer, aliasOpt("d", "del", "rm"),
+	)
+	AddBoolFlag(cmdRunRegistriesDelete, doctl.ArgForce, doctl.ArgShortForce, false, "Delete registry without confirmation prompt")
+	cmdRunRegistriesDelete.Example = `The following example deletes a registry named ` + "`" + `example-registry` + "`" + `: doctl registries delete example-registry. Note that you can delete only one registry at a time.`
+
+	dockerConfigDesc := `Outputs a JSON-formatted Docker configuration that you can use to configure a Docker client to authenticate with your private container registry. This configuration is useful for configuring third-party tools that need access to your registry.
+
+By default this command generates read-only credentials. Use the ` + "`" + `--read-write` + "`" + ` flag to generate credentials that can push. The configuration produced by this command contains a DigitalOcean API token that can be used to access your account and should be treated as sensitive information.`
+
+	cmdRunRegistriesDockerConfig := CmdBuilder(cmd, RunRegistriesDockerConfig, "docker-config <registry-name>",
+		"Generate a Docker auth configuration for a registry",
+		dockerConfigDesc, Writer, aliasOpt("config"))
+	AddBoolFlag(cmdRunRegistriesDockerConfig, doctl.ArgReadWrite, "", false,
+		"Generates credentials that can push to your registry")
+	AddIntFlag(cmdRunRegistriesDockerConfig, doctl.ArgRegistryExpirySeconds, "", 0,
+		"The length of time the registry credentials are valid for, in seconds. By default, the credentials do not expire.")
+	cmdRunRegistriesDockerConfig.Example = `The following example generates a Docker configuration for a registry named ` + "`" + `example-registry` + "`" + ` and uses the ` + "`" + `--expiry-seconds` + "`" + ` to set the credentials to expire after one day: doctl registries docker-config example-registry --expiry-seconds=86400`
+
+	return cmd
+}
+
+// RunRegistriesList lists all registries in the account.
+func RunRegistriesList(c *CmdConfig) error {
+	registries, err := c.Registries().List()
+	if err != nil {
+		return err
+	}
+
+	return displayRegistries(c, registries...)
+}
+
+// RunRegistriesCreate creates a new registry.
+func RunRegistriesCreate(c *CmdConfig) error {
```

Comment:
Should this accept a `subscription-tier` flag like the existing `registry create` command? It does not seem to work as is:

```
$ ./builds/doctl registries -u "https://api.s2r1.internal.digitalocean.com/" -t $DO_STAGE create foo --region s2r1 --trace 
doctl: 2024/11/12 13:13:42 [DEBUG] POST https://api.s2r1.internal.digitalocean.com/v2/registries
doctl: 2024/11/12 13:13:42 -> "POST /v2/registries HTTP/1.1\r\nHost: api.s2r1.internal.digitalocean.com\r\nUser-Agent: doctl/1.117.0-dev (linux amd64) godo/1.129.0\r\nContent-Length: 31\r\nAccept: application/json\r\nContent-Type: application/json\r\nAccept-Encoding: gzip\r\n\r\n{\"name\":\"foo\",\"region\":\"s2r1\"}\n"
doctl: 2024/11/12 13:13:43 [DEBUG] POST https://api.s2r1.internal.digitalocean.com/v2/registries (status: 500): retrying in 1s (5 left)
doctl: 2024/11/12 13:13:44 [DEBUG] POST https://api.s2r1.internal.digitalocean.com/v2/registries (status: 500): retrying in 2s (4 left)
doctl: 2024/11/12 13:13:46 [DEBUG] POST https://api.s2r1.internal.digitalocean.com/v2/registries (status: 500): retrying in 4s (3 left)
doctl: 2024/11/12 13:13:50 [DEBUG] POST https://api.s2r1.internal.digitalocean.com/v2/registries (status: 500): retrying in 8s (2 left)
doctl: 2024/11/12 13:13:59 [DEBUG] POST https://api.s2r1.internal.digitalocean.com/v2/registries (status: 500): retrying in 16s (1 left)
doctl: 2024/11/12 13:14:15 <- "HTTP/2.0 500 Internal Server Error\r\nContent-Length: 59\r\nContent-Type: application/json\r\nDate: Tue, 12 Nov 2024 18:14:15 GMT\r\nRatelimit-Limit: 20000\r\nRatelimit-Remaining: 19982\r\nRatelimit-Reset: 1731438626\r\nX-Gateway: Edge-Gateway\r\nX-Godo-Retry-Attempts: 6\r\nX-Request-Id: 1503e780-ae6e-43d0-8862-114c067e9bc0\r\nX-Response-From: service\r\n\r\n{\"id\": \"Internal Server Error\", \"message\": \"Server Error\" }"
Error: POST https://api.s2r1.internal.digitalocean.com/v2/registries: 500 (request "1503e780-ae6e-43d0-8862-114c067e9bc0") Server Error; giving up after 6 attempt(s)
```

Reply:
You have to be on a `professional` plan for this to work. Right now you are on the `basic` plan. Try upgrading your plan to the professional tier and re-try. 

That said, we need to improve the response message here. Will update the API, shortly.

Reply:
This is the create command. You can't upgrade a registry you can not create.

Reply:
Apologies for the confusion. 😬 

Some context: To create multiple registries (internally: `secondary` registries), the prerequisite is to have a `primary` registry first (which can be created using the existing `v2/registry` endpoint or `doctl registry create` - the current DOCR consumers would already have this created, of course). This has to be done because the `primary` reg is tied to a subscription - when you create the `primary` reg you also create the subscription along with it. The subscription stays bound to the `primary` reg and gets cleaned up once the reg is deleted. 

Coming back to your question: Your `primary` reg has to be on a `professional` plan before you can start creating `secondary` registries (using the new `registries` command/REST API). You already have a `primary` reg, which is on the `basic` plan. You can upgrade it to `professional` either via the cloud panel or the [REST endpoint](https://docs.digitalocean.com/reference/api/api-reference/#operation/registry_update_subscription).